troi/tools/utils.py

```diff
@@ -59,8 +59,36 @@ def playlist_add_tracks(self, playlist_id, track_ids):
 
     def get_playlist_tracks(self, playlist_id):
         url = f"{APPLE_MUSIC_URL}/me/library/playlists/{playlist_id}?include=tracks"
-        response = self.session.get(url, headers=self.headers)
-        return response.json()
+        response = self.session.get(url, headers=self.headers).json()
+        tracks = response["data"][0]["relationships"]["tracks"]["data"]
+        total_tracks_count = response["data"][0]["relationships"]["tracks"]["meta"]["total"]
+        playlist_name = response["data"][0]["attributes"]["name"]
+        playlist_description = response["data"][0]["attributes"].get(
+            "description", {}).get("standard", "")
+
+        # apple music returns only the first 100 tracks with "/playlists/{playlist_id}?include=tracks"
+        # and we need to fetch the rest of the tracks using the "playlists/{playlist_id}/tracks" endpoint
+        if len(tracks) < total_tracks_count:
+            offset = len(tracks)
+            # endpoint returns 100 tracks per call max -> run iteratively with an offset until there are no more tracks
+            while True:
+                url = f"{APPLE_MUSIC_URL}/me/library/playlists/{playlist_id}/tracks?limit=100&offset={offset}"
```

Comment:
Does any more error checking need to be added here? rate limiting or otherwise?

Reply:
We need to ensure that errors that might happen are not silently dropped.

Reply:
Rate limiting an retry strategy are handles by the session
https://github.com/metabrainz/troi-recommendation-playground/blob/6293c4993d40048492e20270269ec546b1b21be9/troi/tools/utils.py#L183C5-L197C16

What should be the strategy for handling other errors?
The API will sometimes return a json body with an `error` key, so I could try parsing for that and logging the error message (instead of the current caught KeyError) as an improvement.
Anything else that I should look out for?